fluss-common/src/main/java/com/alibaba/fluss/record/MemoryLogRecordsArrowBuilder.java

```diff
@@ -227,6 +230,14 @@ public void resetWriterState(long writerId, int batchSequence) {
         this.batchSequence = batchSequence;
     }
 
+    public void overrideLastLogOffset(long lastLogOffset) {
```

Comment:
We don't need to provide this method. we only need to check `recordCount` is 0 or not and write corresponding offset_delta in `writeBatchHeader`. 

It's dangerous to provide this method, as it's only valid if the `lastLogOffset=0 and recordCount=0` or `lastLogOffset=baseOffset+recordCount-1`, but there is no any validation in this method.

Reply:
Done! This method id removed.